source/functions/initializeWecSim.m

```diff
@@ -320,15 +328,19 @@
 
 % Check for elevationImport with nonlinearHydro
 for ii = 1:simu.numHydroBodies
-    if strcmp(waves.type,'elevationImport') && body(ii).nonlinearHydro == 1
-        error(['Cannot run WEC-Sim with nonlinear hydro (body(ii).nonlinearHydro) and "elevationImport" wave type'])
+    for iW = 1:length(waves)
+        if strcmp(waves(iW).type,'elevationImport') && body(ii).nonlinearHydro == 1
+            error(['Cannot run WEC-Sim with nonlinear hydro (body(ii).nonlinearHydro) and "elevationImport" wave type'])
+        end
     end
 end
 
 % Check for elevationImport with morisonElement
```

Comment:
@salhus Minor request: May I ask that you add a check so if there is a mismatch in `waves.type` an error is thrown to stop the simulation. When I tried to break this code with this mismatch the error was not helpful in determining the issue.